apis/Google.Cloud.Logging.NLog/Google.Cloud.Logging.NLog/GoogleStackdriverTarget.cs

```diff
@@ -514,66 +517,19 @@ private LogEntry BuildLogEntry(LogEventInfo loggingEvent)
-                    }
-                }
-                else
-                {
-                    jsonStruct.Fields.Add("message", Value.ForString(RenderLogEvent(Layout, loggingEvent)));
-
-                    if (ServiceContextName != null)
-                    {
-                        var serviceName = RenderLogEvent(ServiceContextName, loggingEvent);
-                        if (!string.IsNullOrEmpty(serviceName))
-                        {
-                            // Include ServiceContext to allow errors to be automatically forwarded
-                            var serviceVersion = RenderLogEvent(ServiceContextVersion, loggingEvent);
-                            if (string.IsNullOrEmpty(serviceVersion))
-                            {
-                                serviceVersion = "0.0.0.0";
-                            }
-
-                            var serviceContext = new Struct();
-                            jsonStruct.Fields.Add("serviceContext", Value.ForStruct(serviceContext));
-                            serviceContext.Fields.Add("service", Value.ForString(serviceName));
-                            serviceContext.Fields.Add("version", Value.ForString(serviceVersion));
-                        }
-                    }
-
-                    var propertiesStruct = new Struct();
-                    jsonStruct.Fields.Add("properties", Value.ForStruct(propertiesStruct));
-
-                    foreach (var combinedProperty in GetAllProperties(loggingEvent).Where(x => !string.IsNullOrEmpty(x.Key)))
-                    {
-                        Value jsonValue;
-                        try
-                        {
-                            jsonValue = _jsonConvertFunction(combinedProperty.Value);
-                        }
-                        catch (Exception ex)
-                        {
-                            InternalLogger.Warn(ex,
-                                "GoogleStackdriver(Name={0}): Exception at BuildLogEntry with Key={1}", Name, combinedProperty.Key);
-                            jsonValue = Value.ForString($"<Exception: '{ex.Message}'>");
-                        }
-
-                        propertiesStruct.Fields.Add(combinedProperty.Key, jsonValue);
-                    }
-                }
-
-                logEntry.JsonPayload = jsonStruct;
+                logEntry.JsonPayload = RenderJsonPayload(loggingEvent);
             }
             else
             {
                 logEntry.TextPayload = RenderLogEvent(Layout, loggingEvent);
 
-                foreach (var combinedProperty in GetAllProperties(loggingEvent).Where(x => !string.IsNullOrEmpty(x.Key)))
+                foreach (var combinedProperty in GetAllProperties(loggingEvent))
                 {
+                    if (string.IsNullOrEmpty(combinedProperty.Key))
+                    {
+                        continue;
+                    }
+
```

Comment:
Hmmm, I like the old way better. I think it's more readable.
Any particular reason you'd prefer this?

Reply:
I really like performance and calling any Linq-expression besides `Any()` or `Count()` performs bonus-class-allocation, even when the collection is empty.